packages/contracts-bedrock/contracts/L1/L2OutputOracle.sol

```diff
@@ -149,32 +145,29 @@ contract L2OutputOracle is OwnableUpgradeable, Semver {
     }
 
     /**
-     * @notice Deletes the most recent output. This is used to remove the most recent output in the
-     *         event that an erreneous output is submitted. It can only be called by the contract's
-     *         owner, not the proposer. Longer term, this should be replaced with a more robust
-     *         mechanism which will allow deletion of proposals shown to be invalid by a fault
-     *         proof.
+     * @notice Deletes all output proposals after and including the proposal that corresponds to
+     *         the given block number. Can only be called by the owner, but will be replaced with
+     *         a mechanism that allows a challenger contract to delete proposals.
      *
-     * @param _proposal Represents the output proposal to delete
+     * @param _l2BlockNumber L2 block number of the first output root to delete.
      */
     // solhint-disable-next-line ordering
-    function deleteL2Output(Types.OutputProposal memory _proposal) external onlyOwner {
-        Types.OutputProposal memory outputToDelete = l2Outputs[latestBlockNumber];
-
+    function deleteL2Outputs(uint256 _l2BlockNumber) external onlyOwner {
+        // Simple check that accomplishes three things:
+        //   1. Prevents deleting anything from before the genesis block.
+        //   2. Prevents deleting anything after the latest block.
+        //   3. Prevents deleting anything other than a checkpoint block.
         require(
-            _proposal.outputRoot == outputToDelete.outputRoot,
-            "L2OutputOracle: output root to delete does not match the latest output proposal"
+            l2Outputs[_l2BlockNumber].outputRoot != bytes32(0),
+            "L2OutputOracle: cannot delete a non-existent output"
```

Comment:
In the case that an output has already been deleted, check number 2 above is not necessarily true. 

We also need to prevent deleting finalized outputs.

Reply:
> check number 2 above is not necessarily true.

To be clear, I think that's OK. We just need to make sure our comments are accurate.

Reply:
Fixed the first issue. I don't know how to best handle the second issue because the finalization period is part of the portal and not the output oracle. It may not actually make sense to do the check in the oracle because e.g., if we replaced the proof system with a zk proof then there is no sense of "finalization".

Reply:
Right, an output is either valid or not regardless of age. SGTM.